src/ClaimToken.sol

```diff
@@ -0,0 +1,205 @@
+
+        _isEventOngoing[tokenAddress][eventIDHash] = true;
+        _isEventCreated[tokenAddress][eventIDHash] = true;
+        emit EventCreated(tokenAddress, eventID);
+    }
+
+    // Updates the status of an existing event
+    function updateEvent(address tokenAddress, string calldata eventID, bool isEventClosed)
+        external
+        override
+        onlyOwner
+    {
+        bytes32 eventIDHash = _hashString(eventID);
+
+        require(_isEventCreated[tokenAddress][eventIDHash], "Event ID for the token not created yet");
+
+        require(_isEventOngoing[tokenAddress][eventIDHash] != !isEventClosed, "Event already in this state");
+
+        _isEventOngoing[tokenAddress][eventIDHash] = !isEventClosed;
+        emit EventUpdated(tokenAddress, eventID, isEventClosed);
+    }
+
+    // Claims tokens for a user based on the provided data
+    function claim(
+        address tokenAddress,
+        string calldata eventID,
+        address userAddress,
+        uint256 amount,
+        bytes calldata signerSignature
+    ) external override nonReentrant {
+        bytes32 eventIDHash = _hashString(eventID);
+        require(_isEventOngoing[tokenAddress][eventIDHash], "Event is closed");
+
+        if (_userClaimedAmount[tokenAddress][eventIDHash][userAddress] != 0) {
+            revert UserAlreadyClaimedToken(userAddress);
+        }
+
+        bytes32 claimHash = getClaimHash(tokenAddress, eventID, userAddress, amount);
+        bytes32 ethSignedMessageHash = MessageHashUtils.toEthSignedMessageHash(keccak256(abi.encode(claimHash)));
+        address signer = ECDSA.recover(ethSignedMessageHash, signerSignature);
+
+        require(_isActivatedSigner[signer], "Invalid signer");
+
+        SafeERC20.safeTransfer(IERC20(tokenAddress), userAddress, amount);
+
+        _userClaimedAmount[tokenAddress][eventIDHash][userAddress] += amount;
+
+        emit Claimed(claimHash, tokenAddress, eventID, userAddress, amount, signerSignature);
+    }
+
+    // Transfers tokens from the contract to a recipient
+    function transferToken(address tokenAddress, address recipient, uint256 amount) external onlyOwner {
+        require(tokenAddress != address(0), "Invalid token address");
+        require(recipient != address(0), "Invalid recipient address");
+
+        IERC20(tokenAddress).transfer(recipient, amount);
+    }
+
+    // ------------------------
+    // -- Internal Functions --
+    // ------------------------
+
+    function _hashString(string memory input) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(input));
+    }
+
+    // -----------------------
+    // -- Private Functions --
+    // -----------------------
+
+    // Updates signers and their activation status
+    function _updateSigners(address[] memory _signerList, bool[] memory _isActivatedList) private {
+        require(_signerList.length == _isActivatedList.length, "Mismatch in input lengths");
```

Comment:
If we are using custom error, then we should use custom error throughout.

Reply:
Try using "require + custom error": https://soliditylang.org/blog/2024/09/04/solidity-0.8.27-release-announcement/

Reply:
Nice!
In addition to converting the current 6 `require` statements to the `require(xxx, Error())` format,
should we also apply the same change to the other `if(xxx) revert Error()` cases, converting them to the `require(xxx, Error())` format?

Reply:
Yes